File: Data Structures & Algorithms/valid-palindrome-ii/submission-1.cpp
class Solution {
public:
    bool validPalindrome(string s) {
        int l = 0, r = s.size() - 1;
        int count = 1;
        while (l < r) {
            if (s[l] == s[r]) {
                l++; r--;
            } else if (count--){
                if (s[l] == s[r - 1]) {
                    r--;
                } else if (s[l + 1] == s[r]) {
                    l++;
                } else {
                    return false;
                }
            } else {
                return false;
            }
        }
        return true;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        countMap = {}

        for num in nums:
            countMap[num] = 1 + countMap.get(num, 0)
        
        freqMap = [[] for _ in range(len(nums) + 1)]
        for num, count in countMap.items():
            freqMap[count].append(num)
        
        res = []
        for i in range(len(freqMap) - 1, 0, -1):
            if len(freqMap[i]) > 0:
                res.extend(freqMap[i])
                if len(res) == k:
                    break
        return res
